### ISSUE TITLE
False positive when adding CharField(blank=True) with implicit empty string

### ISSUE THREAD

**Issue by johnnymetz** [3YOURMIND/django-migration-linter#310]
When adding a `CharField(..., blank=True)` (without a default) to a model, Django's migration system automatically generates an empty string default in the SQL.

However, django-migration-linter still flags this migration as failing the "NOT NULL" check which is a false positive in this context.

**Steps to reproduce**

- Add a new blankable CharField to a Django model:

```
my_field = models.CharField(max_length=50, blank=True)
```

- Run makemigrations
- Notice generated sql includes a default of an empty string

```
python manage.py sqlmigrate app 0004

# Output
ALTER TABLE "api_organization" ADD COLUMN "my_field" varchar(50) DEFAULT '' NOT NULL;
ALTER TABLE "api_organization" ALTER COLUMN "my_field" DROP DEFAULT;
COMMIT;
```

- Linter flags it as unsafe

**Expected behavior**

The linter should not flag it as unsafe.

**Environment**

- Django version: 5.2.4
- postgres: version 15.13
- django-migration-linter version: 5.2.0

**Notes**

- `blank=True` is key here. Without it, the migration doesn't automatically add the empty string default value.
- The empty string default is also set when creating an instance: `Organization.objects.create()` works fine and my_field is set to an empty string. However, when `blank=False`, the create fails with the following error: `django.core.exceptions.ValidationError: {'my_field': ['This field cannot be blank.']}`
- I don't see this documented in the Django docs anywhere

**Comment by johnnymetz** [comment]
Nevermind this is indeed a backwards incompatible migration because the db default is removed so writes on the blue server will fail during a rolling deployment.
